channel.go

```diff
@@ -418,6 +419,7 @@ It is safe to call this method multiple times.
 
 */
 func (ch *Channel) Close() error {
+	defer ch.connection.releaseChannel(ch.id)
```

Comment:
You are right. I'm going to see if we can achieve the same effect without using `defer` (nothing wrong with it, just easier to reason about things without delayed operations).